electrum/address_synchronizer.py

```diff
@@ -327,6 +329,13 @@ def remove_from_spent_outpoints():
             self.db.remove_txi(tx_hash)
             self.db.remove_txo(tx_hash)
             self.db.remove_tx_fee(tx_hash)
+            self.db.remove_verified_tx(tx_hash)
+            self.unverified_tx.pop(tx_hash, None)
+            if tx:
```

Comment:
when is tx None?

Reply:
Hmmm I guess if the wallet is not corrupted then it should never happen. Seems hard to come up with a case where we would try to call `remove_transaction` before `add_transaction` for a given tx.
I had decided to handle the case simply because it is already considered possible inside remove_transaction:
https://github.com/spesmilo/electrum/blob/cfbd83c43230ad6b31e98120daff8336c3ac0b7a/electrum/address_synchronizer.py#L303-L305

Also, there is `remove_local_transactions_we_dont_have`, which if gets called it would certainly trigger this.
Then again, that method should only get called for (partially) corrupted wallet I think:
https://github.com/spesmilo/electrum/commit/1935d7d1654b7b90f35167e342bcfc0732c69bce
https://github.com/spesmilo/electrum/issues/4232